src/core/geometry/qgspointv2.cpp

```diff
@@ -464,21 +464,33 @@ double QgsPointV2::azimuth( const QgsPointV2& other ) const
 }
 
 
-QgsPointV2 QgsPointV2::project( double distance, double azimuth, double inclinaison ) const
+QgsPointV2 QgsPointV2::project( double distance, double azimuth, double inclination ) const
 {
+    QgsWkbTypes::Type pType(QgsWkbTypes::Point);
+
     double rads_xy = azimuth * M_PI / 180.0;
-    if ( !is3D() )
+    double dx = 0.0, dy = 0.0, dz = 0.0;
+
+    inclination = fmod( inclination, 360.0 );
+
+    if ( !is3D() && inclination == 90.0 )
```

Comment:
I think it would be slightly safer to do:

    if ( !is3D() && qgsDoubleNear( inclination, 90.0 ) )

qgsDoubleNear allows for a tiny tolerance to account for floating point calculations.

I'm also wondering... what about an inclination of 270? Probably an extreme corner case, but worth covering too.